src/select/select.js

```diff
@@ -384,6 +387,12 @@ class Select extends React.Component<PropsT, StatelessStateT> {
     return type === TYPE.search ? true : multiple;
   }
 
+  openDropDown(newTextValue?: string, callback?: () => void) {
+    this.loadOptions(newTextValue).then(() => {
+      this.setState({isDropDownOpen: true}, callback);
```

Comment:
There will probably be cases where async results take 1 or 2 seconds to return, and with this logic the dropdown may seem broken or laggy.

We should probably open the dropdown without waiting for results and show a spinner in the dropdown until they're finished loading.

Reply:
so different products may have full loading mask or smaller one for particular region\component. We don't know the exact use case. Would you like me to use our standard spinner?

Reply:
+1 previous conversation can be [found here](https://github.com/uber-web/baseui/pull/323#discussion_r221406151)

Reply:
ok, fixed, added spinner